File: Programmers/Lv3/인사고과.py
# 근태점수, 동료평가 점수
# 두 점수의 합이 높은 순 석차

def solution(scores):
    answer = 1
    me = scores[0]
    scores.sort(key=lambda x : (-x[0], x[1]))
    maxB = 0

    for a, b in scores:
        if me[0] < a and me[1] < b:
            return -1

        if b >= maxB:
            maxB = b
            if a + b > sum(me):
                answer += 1
    return answer

File: Programmers/Lv3/인사고과.js
function solution(scores) {
  var me = scores[0];
  scores = scores.sort((a, b) => b[0] - a[0] || a[1] - b[1]);
  let maxB = 0;
  var cnt = 1;
  for (let [a, b] of scores) {
    if (a > me[0] && b > me[1]) {
      return -1;
    }

    if (b >= maxB) {
      maxB = b;
      if (a + b > me[0] + me[1]) {
        cnt++;
      }
    }
  }

  return cnt;
}
